crates/sargon/src/wrapped_radix_engine_toolkit/low_level/v2/transaction_manifest_v2/execution_summary_v2/transaction_manifest_execution_summary_v2.rs

```diff
@@ -0,0 +1,69 @@
+use crate::prelude::*;
+
+impl TransactionManifestV2 {
+    /// Creates the `ExecutionSummary` based on the `engine_toolkit_receipt`.
+    ///
+    /// Such value should be obtained from the Gateway `/transaction/preview` endpoint, under the `radix_engine_toolkit_receipt` field.
+    pub fn execution_summary(
+        &self,
+        engine_toolkit_receipt: ScryptoSerializableToolkitTransactionReceipt,
+    ) -> Result<ExecutionSummary> {
+        let network_definition = self.network_id().network_definition();
+        let runtime_receipt = engine_toolkit_receipt
+            .into_runtime_receipt(&ScryptoAddressBech32Decoder::new(
+                &network_definition,
+            ))
+            .ok()
```

Comment:
hmm double `ok()`? probably you can `map_err(|e| { error!("Failed to bla bla {:?}", e); CommonError::Fail })` instead?

Reply:
Agree. Updated in `transaction_manifest_execution_summary.rs` too